dsync/http.go

```diff
@@ -0,0 +1,232 @@
+package dsync
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/qri-io/dag"
+)
+
+// HTTPClient is the request side of doing dsync over HTTP
+type HTTPClient struct {
```

Comment:
Would be great to have some mention that this satisfies the remote (or whatever we rename it) interface.
also
`var _ Remote = (*HTTPClient)(nil)`